### PR TITLE
Use FTP-esque implementation to facilitate file transfers

### PR DESCRIPTION
**Pull request by TonyxSun**
This pull request introduces a new FTP-based file transfer feature and makes several modifications to the `TaskRequest` class to support file-based training data. The most important changes include adding methods for file handling and FTP operations, updating the `TaskRequest` class to handle training files, and modifying the networking components to support FTP.

* Added trainingFile field and associated helpers to task_request. 
*  Added FTP specific socket connection functions in utility

**Workflow**:
On the requester side, ensure that a trainingFile is specified in the TaskRequest object. Then, send the TaskRequest object as usual. The requester then awaits for a request/reply (this is unchanged). If the reply message is `get …`, then it assumes FTP *server* role (more on this).

The provider receives the TaskRequest object. If the trainingFile field is non-empty, it will assume FTP based requests. It will call the `getFileFTP` function on its server (note: this acts as the FTP *client*, which requests for a file). 

The getFileFTP func will request and download the specified filename to the provider side (FTP *client side*) and return. It interacts with the FTP *server* side, which is the requestor that has the source data file. 

Then, the training data is read and the provider proceeds as usual. 

**Demo:**
https://youtu.be/sPjZaLmHDw0 

**Note:** 
The underlying network flow is implemented through TCP based data transfers.

### CODE REVIEW

File: include/RequestResponse/task_request.h [hunk 1]
```diff
@@ -21,11 +26,16 @@ class TaskRequest : public Payload {
     void setLeaderUuid(const std::string& leaderUuid);
     void setAssignedWorkers(const AddressTable& assignedWorkers);
     void setTrainingData(const std::vector<int>& trainingData);
+    void setTrainingFile(const std::string& trainingFile);
+    void setTrainingDataFromFile();
+
+    void createTrainingFile(const std::string& newTrainingFileName);
```
Comment (PalaashKolhe) [inline include/RequestResponse/task_request.h:32]: `createTrainingFile` should only be run once. would it make more sense to make this a private function and have it be called on `TaskRequest` initialization?

File: include/utility.h [hunk 2]
```diff
@@ -6,6 +6,15 @@
...
 #include <unordered_map>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <arpa/inet.h>
+
+#define MAXLINE 4096 /*max text line length*/
```
Comment (PalaashKolhe) [inline include/utility.h:17]: a bit confused on what this sets the MAX for. does it set the MAX limit for a training request data object (the broken up packets)?

File: src/Networking/client.cpp [hunk 3]
```diff
@@ -74,6 +72,58 @@ int Client::sendRequest(const char* data) {
...
+    /*
+     * If the request received contains the keyword "get", which is used to
+     * represent a file transfer request, the client will proceed to provide the
+     * file.
+     */
+    char *token, *dummy;
+    dummy = buffer;
+    token = strtok(dummy, " ");
+    // process file descriptor
+    if (strcmp("get", token) == 0) {
```
Comment (PalaashKolhe) [inline src/Networking/client.cpp:85]: would this comparison change when we add the protobuf serialization?

File: include/utility.h (hunk 2 above)
Comment (TonyxSun) [reply include/utility.h:17]: Yea, perhaps naming as `BUFFER_SIZE` is more appropriate. The usage of this const is a little overloaded.

File: src/Networking/client.cpp (hunk 3 above)
Comment (TonyxSun) [reply src/Networking/client.cpp:85]: It wouldn't. This is part of the system command that tells the "FTP server" it wants a file (ie. `get trainingData.proto`). We will have to change everything that marshals from a file into `vector<int>` though. 

```
/*
     * If the request received contains the keyword "get", which is used to
     * represent a file transfer request, the client will proceed to provide the
     * file.
     */
     
```

File: include/RequestResponse/task_request.h (hunk 1 above)
Comment (TonyxSun) [reply include/RequestResponse/task_request.h:32]: Good idea  👍🏼

**Comment by TonyxSun** [comment]
Thanks Palaash for the comments earlier

**Review by PalaashKolhe — APPROVED** [review]
LGTM 🚀

**Pull request merged**